src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,45 @@
 
 class Animal {
   // write your code here
+  health = 100;
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  }
+  bite(herbivore) {
+    if (!herbivore.hidden && herbivore instanceof Herbivore) {
+      herbivore.health -= 50;
+
+      if (herbivore.health === 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal !== herbivore);
+      }
```

Comment:
[CODE PERFORMANCE] - Instead of checking if the health is exactly 0, it's better to check if the health is less than or equal to 0 when filtering the `Animal.alive` array. This will ensure that any animal with non-positive health will be removed, covering edge cases where health might drop below 0 due to future changes in the bite logic.